src/Catalyst.Core.Lib/Cryptography/CertificateStore.cs

```diff
@@ -0,0 +1,199 @@
+                    Environment.NewLine +
+                    "cf. `https://github.com/catalyst-network/Catalyst.Node/wiki/Creating-a-Self-Signed-Certificate` for instructions");
+            }
+
+            certificate = CreateAndSaveSelfSignedCertificate(pfxFilePath);
+
+            return certificate;
+        }
+
+        private X509Certificate2 CreateAndSaveSelfSignedCertificate(string filePath, string commonName = LocalHost)
+        {
+            const string promptMessage = "Catalyst Node needs to create an SSL certificate." +
+                " Please enter a password to encrypt the certificate on disk:";
+
+            using (var password = _passwordManager.RetrieveOrPromptAndAddPasswordToRegistry(_certificatePasswordIdentifier, promptMessage))
+            {
+                var certificate = BuildSelfSignedServerCertificate(password, commonName);
+                Save(certificate, filePath, password);
+                return certificate;
+            }
+        }
+
+        private void Save(X509Certificate certificate, string fileName, SecureString password)
+        {
+            var targetDirInfo = _storageFolder;
+            if (!targetDirInfo.Exists)
+            {
+                targetDirInfo.Create();
+            }
+
+            var certificateInBytes = certificate.Export(X509ContentType.Pfx, password);
+            var fullPathToCertificate = Path.Combine(targetDirInfo.FullName, fileName);
+            File.WriteAllBytes(fullPathToCertificate, certificateInBytes);
+
+            Logger.Warning("A certificate file has been created at {0}.",
+                fullPathToCertificate);
+            Logger.Warning("Please make sure this certificate is added to " +
+                "your local trusted root store to remove warnings.", fullPathToCertificate);
+        }
+
+        private bool TryGet(string fileName, out X509Certificate2 certificate)
+        {
+            var fullPath = Path.Combine(_storageFolder.ToString(), fileName);
+            var fileInfo = new FileInfo(fullPath);
+            certificate = null;
+
+            if (!fileInfo.Exists)
+            {
+                return false;
+            }
+
+            try
+            {
+                var fileInBytes = File.ReadAllBytes(fullPath);
+
+                var passwordPromptMessage =
+                    $"Please type in the password for the certificate at {fullPath} (optional):";
+                SecureString passwordFromConsole = null;
+                while (PasswordTries < MaxTries)
+                {
+                    try
+                    {
+                        passwordFromConsole =
+                            _passwordManager.RetrieveOrPromptAndAddPasswordToRegistry(_certificatePasswordIdentifier, passwordPromptMessage);
+                        certificate = new X509Certificate2(fileInBytes, passwordFromConsole);
+
+                        break;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor the code in order to remove this break statement.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)